estate/models/estate_property.py

```diff
@@ -0,0 +1,86 @@
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+from datetime import date, timedelta
+
+
+class EstateProperty(models.Model):
+    _name = "estate_property"
+    _description = "An Estate Property"
+    _order = "id DESC"
+
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, default=date.today() + timedelta(days=90))
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer(string="Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        [
+            ('north', 'North'),
+            ('west', 'West'),
+            ('east', 'East'),
+            ('south', 'South')
+        ]
+    )
+    active = fields.Boolean("Active", default=True)
+    state = fields.Selection([
+        ('new', 'New'),
+        ('received', 'Offer Received'),
+        ('accepted', 'Offer Accepted'),
+        ('sold', 'Sold'),
+        ('cancelled', 'Cancelled')],
+        default='new',
+        required=True
+    )
+    property_type_id = fields.Many2one('property_type', string='Type')
+    user_id = fields.Many2one('res.users', string='Salesperson', default=lambda self: self.env.user)
+    buyer_id = fields.Many2one('res.partner', copy=False)
+    tags_ids = fields.Many2many("property_tag", string="Tags")
+
+    offer_ids = fields.One2many("property_offer", "property_id", string="Offers")
+    total_area = fields.Float(compute="_compute_area")
+    best_offer = fields.Float(compute="_compute_best_offer")
+
+    @api.depends("garden_area", "living_area")
+    def _compute_area(self):
+        for record in self:
+            record.total_area += record.garden_area + record.living_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        for record in self:
+            prices = record.offer_ids.mapped('price')
+            record.best_offer = max(prices) if prices else 0.0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = None
+
+    def action_set_cancelled(self):
+        for record in self:
+            if record.state == 'sold':
+                raise UserError("Why you wanna cancel something sold fam?")
```

Comment:
Some strings are automatically translated but this is not the case for error messages. To make it translatable, use `_()` method 👍
```suggestion
                raise UserError(_("Why you wanna cancel something sold fam?"))
```